### ISSUE TITLE
Meaning behind --update-mean-var --train-beta-gamma

### ISSUE THREAD

**Issue by Tamme** [hellochick/ICNet-tensorflow#40]
Hi.

I havent encountered in other projects this kind of updating values. Is it something originating from PSPNet or this really is the way to use momentum or is it something third?

Thanks,
Tamme

**Comment by hellochick** [comment]
Hey @Tamme, let me explain these two flags first. There are four variables in the batch normalization layer, and they are moving_mean, moving_variance, gamma, and beta respectively. And moving_mean and moving variance are not trainable variables, so we need to update them using update ops which is put in `tf.GraphKeys.UPDATE_OPS`, you can take a look at [tensorflow docs](https://www.tensorflow.org/api_docs/python/tf/layers/batch_normalization). So I use the flag `--update-mean-var` to decide whether to update mean and var ( Since update in large batch size is better, if we train in mini-batch,  we can frozen these two variables ).

**Comment by manuel-88** [comment]
hey, when I run the training without `-update-mean-var` the evaluation results are almost zero. Do you know why @hellochick ?

**Comment by hellochick** [comment]
@manuel-88, if you didn't update mean and variance, then the batch normalization layer will do nothing. Maybe this is the problem.
